### PR TITLE
Ensure failoverCandidates have no servers not in Plan.

### PR DESCRIPTION
**Pull request by neunhoef**
Fix failoverCandidates for servers not in Plan.

Problem: we must never fail over to a server which is not in Plan
because it might not have the data.

TODO:

 - [X] AddFollower, FailedFollower, CleanOutServer and MoveShard will not
       add a server to the Plan which is in failoverCandidates for this
       shard or any of the ones with distributeShardsLike.
 - [X] add preconditions to ensure that it is snapshot-safe
 - [X] make dbserver Maintenance remove servers that are not in Plan from
       failoverCandidates.

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12191/

File: arangod/Cluster/Maintenance.cpp [hunk 1]
```diff
@@ -244,19 +244,32 @@ void handlePlanShard(StorageEngine& engine, uint64_t planIndex, VPackSlice const
...
+        if (shards.isObject()) {
+          VPackSlice planServers = shards.get(shname);
+          if (planServers.isArray()) {
             std::unordered_set<std::string> followersToDrop;
-            for (auto const& q : VPackArrayIterator(inSyncFollowers)) {
-              followersToDrop.insert(q.copyString());
+            // Now we have two server lists (servers and
+            // failoverCandidates, we are looking for a server which
+            // occurs in either of them but not in the plan
+            VPackSlice serverList = lcol.get("servers");
```
Comment (mchacki) [inline arangod/Cluster/Maintenance.cpp:259]:
```suggestion
            VPackSlice serverList = lcol.get(SERVERS);
```

**Review by mchacki — CHANGES_REQUESTED** [review]
Code LGTM.
But we are loosing one static string here, see my suggestion

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12198/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12200/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12203/

**Review by mchacki — APPROVED** [review]
LGTM

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12220/

**Comment by mchacki** [comment]
Retry Jenkins after devel merge
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12531/

**Comment by mchacki** [comment]
The red test in Jenkins seems unrelated, I am investigating it now.
All other tests blue.

**Comment by mchacki** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12540/
Jenkins Blue, red test above is unrelated we work on it separately

**Pull request merged**